tests/vm_delete_protection_test.go

```diff
@@ -58,6 +57,11 @@ var _ = Describe("VM delete protection", func() {
 	)
 
 	DescribeTable("should be able to delete a VM if the protection is disabled", func(labelValue string) {
+		DeferCleanup(func() {
```

Comment:
No need to use `DeferCleanup` here. The `waitForDeletion` can be called after the `Expect(apiClient.Delete(ctx, vm)).To(Succeed())`. Otherwise it would wait for deletion that did not happen.

Btw combining `DeferCleanup()` and `AfterEach()` in a single test case is not a good idea, because their ordering does not follow tree structure. First all `AfterEach()` are executed and then all `DeferCleanup()`.